huffman/src/table.hpp

```diff
@@ -67,18 +67,20 @@ class table
 
   constexpr auto encode_symbols() -> void
   {
+    auto reversed = std::views::reverse(table_);
+
     // precondition, audit
```

Comment:
what does the comment `audit` mean here?

Reply:
The terminology comes from the proposals for contracts (which haven't been added to the standard yet):
https://herbsutter.com/2018/07/02/trip-report-summer-iso-c-standards-meeting-rapperswil/

I originally wrote this as checking that the input data is sorted in functions that require sorted data is expensive (e.g. binary search has log time complexity but checking data is sorted has linear time complexity). 

However, although we require the data to be sorted, the construction algorithm is still linear time as we need to visit every node.